graphql/client.go

```diff
@@ -177,7 +249,73 @@ func (c *client) MakeRequest(ctx context.Context, req *Request, resp *Response)
 	return nil
 }
 
+func (w *webSocketClient) DialWebSocket(ctx context.Context, req *Request, respChan chan json.RawMessage) (errChan chan error, err error) {
+	if req.Query != "" {
+		if strings.HasPrefix(strings.TrimSpace(req.Query), "query") {
+			return nil, errors.New("client does not support queries")
+		}
+		if strings.HasPrefix(strings.TrimSpace(req.Query), "mutation") {
+			return nil, errors.New("client does not support mutations")
+		}
+	}
+
+	err = w.subscribeAndListen(
+		ctx,
+		req,
+		respChan,
+	)
+
+	return w.errChan, err
+}
+
+func (w *webSocketClient) CloseWebSocket() {
+	defer w.conn.Close()
+	err := w.sendComplete()
+	if err != nil {
+		w.errChan <- err
+	}
+	err = w.conn.WriteMessage(closeMessage, formatCloseMessage(closeNormalClosure, ""))
+	if err != nil {
+		w.errChan <- err
+	}
+}
+
+func (w *webSocketClient) subscribeAndListen(ctx context.Context, req *Request, respChan chan json.RawMessage) error {
+	var err error
+	w.conn, err = w.Dialer.DialContext(ctx, w.endpoint, w.Header)
+	if err != nil {
+		return err
+	}
+
+	err = w.sendInit()
+	if err != nil {
+		w.conn.Close()
+		return err
+	}
+	err = w.waitForConnAck()
+	if err != nil {
+		w.conn.Close()
+		return err
+	}
+
+	go w.listenWebSocket(respChan)
```

Comment:
So I guess we end up with two goroutines per subscription, one in the `graphql.Client` to wait on the websocket client and forward to `respChan`, and then one from the generated code that waits on `respChan` and forwards to `dataChan` and `errChan`. Can we avoid that? (Or at least, can we set up the interface so it's possible to avoid in the future, even if it's not worth the effort now?)